### PR TITLE
bank: make delay in depositAll and depositAllExcept optional

### PR DESCRIPTION
**Pull request by yuri-moens**

### CODE REVIEW

**Comment by buracc** [comment]
Not a fan of these types of customisations in the api tbh. It should be up to the plugin developer to implement a custom delay in their plugin/logic.

**Comment by yuri-moens** [comment]
Sure, although I think depositing multiple items with depositAll will trigger an InteractionException if the user is using mouse events because it's going too fast. Should I rework depositAll to only deposit one item again and get rid of depositAllExcept? That way people can control the delay between deposits.

**Comment by buracc** [comment]
Interacting too fast shouldn't be taken into account when developing api methods. An api method should only execute one interaction for this reason. 

depositAll and depositAllExcept should the first item it finds with the given predicate, not all of the items. The plugin developer should handle depositing multiple items themselves. This way you also give the developer the freedom of customizing the delay inbetween interactions.

File: runelite-client/src/main/java/dev/hoot/api/items/Bank.java [hunk 1]
```diff
@@ -175,89 +175,19 @@ public static boolean isEmpty()
...
-			{
-				deposit(item.getId(), Integer.MAX_VALUE);
-				Time.sleepTick();
-			});
-		}
-	}
-
-	public static void depositAllExcept(String... names)
-	{
-		depositAllExcept(x -> Arrays.stream(names).anyMatch(name -> x.getName().equals(name)));
```
Comment (buracc) [inline runelite-client/src/main/java/dev/hoot/api/items/Bank.java:232]: You don't need to remove depositAllExcept, it's useful as a shorthand for depositAll with a negated filter

**Pull request merged**